ffi/saltyrtc_task_relayed_data_ffi.h

```diff
@@ -327,6 +352,18 @@ typedef struct salty_keypair_t salty_keypair_t;
  */
 typedef struct salty_remote_t salty_remote_t;
 
+/*
+ * The return value when encrypting or decrypting raw data.
+ *
+ * Note: Before accessing `bytes`, make sure to check the `success` field for
+ * errors. If an error occurred, the other fields will be `null`.
+ */
+typedef struct {
+  salty_client_encrypt_decrypt_success_t success;
+  const uint8_t *bytes;
+  size_t bytes_len;
+} salty_client_encrypt_decrypt_ret_t;
```

Comment:
A few notes:

In C, I would always return a single error code and let other values you would like to *return* be written by dereferencing further arguments. This also avoids the question on what to set the values to in case an error occurred. But if that is inconsistent with the rest of the API, leave it.

No need to typedef (again, ignore if it's inconsistent with the rest of the API...)

Reply:
> In C, I would always return a single error code and let other values you would like to return be written by dereferencing further arguments.

I'm not sure how practical that is, most data is (and should be) allocated by Rust, so that pattern won't work in all cases.

> No need to typedef

Bindings are generated, so I can't influence that.

Reply:
Allocating data is still easily possible in the same way as you do it now. It simply would be a `uint8_t** bytesp` and then you dereference it to set the pointer to the data.

Reply:
That's possible. In any case, that would require updating the entire API which isn't worth it. The current approach is consistent with the rest of the API :)

Reply:
Then leave it. :)